### ISSUE TITLE
Does eval_metric affect model training?

### ISSUE THREAD

**Issue by Gazillionth** [dmlc/mxnet#1915]
Hi,

The following reflects my understanding of training a network with MXNet. I'd appreciate it if someone could let me know if it's not correct:
- The final **output layer of a network defines the loss function** with respect to which the network will be optimized, e.g. using mxnet.symbol.LogisticRegressionOutput means that the cross-entropy is being minimized (at least, I suppose it's cross entropy, but one way or another, this symbol defines the loss function).
- The **eval_metric** argument supplied to the model.fit function serves only to report to the user the performance of the model with respect to some prediction task (e.g. like top-5 correct), but it does not affect the training process at all.

Is this correct? Or does eval_metric somehow define the loss function that is optimized during training?

Thanks,

Gaz.

**Comment by piiswrong** [comment]
It's just for reporting

**Comment by Gazillionth** [comment]
Thanks, good to have that confirmed. As a follow-up question, that must mean that when training with a single set of hyperparameters, the validation set is only for reporting too, right?

**Comment by pluskid** [comment]
Yes, validation set is only for reporting. Except you use some customized learning rate scheduler that adjust the learning rate according to the performance on the validation set, etc.
